Sources/Pool.swift

```diff
@@ -0,0 +1,7 @@
+public protocol Pool {
+    associatedtype Poolable
+    func borrow() -> Poolable?
+    func takeBack(poolable: Poolable)
+    func with<Type>(handler: (poolable: Poolable) throws -> Type?) throws -> Type?
```

Comment:
This has no meaning, I think it should be removed. I don't see the value to this function, if it does have value, the name needs to be more clear. We should reserve `with` to more often than not follow an action, ie: `runWith`, `moveWith`. Having just `with` doesn't provide context on behavior.

My guess is that it's similar to a map and instead of `borrow` user would call `with(` a handler that should convert. If that's the case, I don't think it should be part of the protocol. The logic is pretty clear. Would you mind describing expected behavior and we can decide as a group if it should be a part of the protocol.